### PR TITLE
Refactor inline widget code into it's own file/class: ProblemWidget.

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
@MiguelCastillo Please review -- for a Brackets source code example, see https://github.com/adobe/brackets/blob/master/src/extensions/default/WebPlatformDocs/InlineDocsViewer.js.

### CODE REVIEW

File: ProblemWidget.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        this.line = line;
+
+        this._sizeEditorToContent   = this._sizeEditorToContent.bind(this);
+    }
+
+    ProblemWidget.prototype = Object.create(InlineWidget.prototype);
+    ProblemWidget.prototype.constructor = ProblemWidget;
+    ProblemWidget.prototype.parentClass = InlineWidget.prototype;
+
+    ProblemWidget.prototype.$wrapperDiv = null;
```
Comment (MiguelCastillo) [inline ProblemWidget.js]: You dont need to define properties in the prototype unless they are considered statically accessible

**Comment by MiguelCastillo** [comment]
Sorry for my neglect on this. Is this fixing an actual bug or just code cleanup?

File: ProblemWidget.js (hunk 1 above)
Comment (Mark-Simulacrum) [inline ProblemWidget.js]: Copied from Brackets code; will change.

**Comment by Mark-Simulacrum** [comment]
Code cleanup, no bugs fixed (that I am aware of).

File: ProblemWidget.js (hunk 1 above)
Comment (MiguelCastillo) [inline ProblemWidget.js]: @Mark-Simulacrum I am going to merge this in.  Do we need `$wrapperDiv` and `$scroller` in the prototype?

**Pull request merged**